### PR TITLE
Template docs, add vault PKI integration example.

### PR DESCRIPTION
**Pull request by csawyerYumaed**
Update the template documentation, and add a vault PKI integration example.

### CODE REVIEW

File: website/source/docs/job-specification/template.html.md [hunk 1]
```diff
@@ -223,6 +223,24 @@ will be identical to the contents of the file.
 For more details see [go-envparser's
 README](https://github.com/schmichael/go-envparse#readme).
 
+## Vault Integration
+
+This example will grab a PKI certificate from Vault and put it out on local disk for your application in PEM format
+including the CA, public and private key in 1 file.
+
```
Comment (preetapan) [inline website/source/docs/job-specification/template.html.md]: This example grabs a PKI certificate from Vault in pem format and stores it into your application's local directory.

Comment (csawyerYumaed) [reply website/source/docs/job-specification/template.html.md]: This is not actually true.  it's $${NOMAD_SECRETS_DIR}, which would become /secrets/bundle.pem not the /local/ dir, which would be: $${NOMAD_TASK_DIR}

Comment (csawyerYumaed) [reply website/source/docs/job-specification/template.html.md]: I'm fine changing it to: "This example grabs a PKI certificate from Vault in pem format and stores it into your application's secret directory."

You are welcome to pull and change directly and not wait on me to figure GitHub out and do another commit and pull request update.

**Comment by schmichael** [comment]
Fixed up a few last little bits like the double $$s. Thanks @csawyerYumaed!

**Pull request merged**